prompting-client/src/snapd_client/interfaces/home.rs

```diff
@@ -457,6 +489,24 @@ impl<'a> CategorisedPath<'a> {
             _ => None,
         }
     }
+
+    fn get_top_level_dir(&self) -> String {
+        debug_assert!(matches!(
+            self.kind,
+            PathKind::TopLevelDir | PathKind::TopLevelDirFile
+        ));
+        let top_level: PathBuf = self.path.iter().take(1).collect();
+        format!("{}", top_level.to_string_lossy())
```

Comment:
you should be able to do the following here I think?
```rust
top_level.to_string_lossy().to_owned()
```

Rather than needing to go through the format macro